Userland/Libraries/LibJS/Runtime/DateConstructor.cpp

```diff
@@ -176,22 +176,43 @@ Value DateConstructor::construct(Function&)
         auto milliseconds = static_cast<u16>(tv.tv_usec / 1000);
         return Date::create(global_object(), datetime, milliseconds);
     }
+
+    auto invalid_arg = [&vm](size_t i) { return vm.argument(i).is_nan() || vm.argument(i).is_undefined(); };
```

Comment:
This is incorrect, the arguments should be checked after being coerced to numbers - e.g. `new Date("")` is not matched by this but returns an invalid date.